src/python/pants/core/goals/export.py

```diff
@@ -118,6 +124,15 @@ async def export(
     all_results = await MultiGet(Get(ExportResults, ExportRequest, request) for request in requests)
     flattened_results = [res for results in all_results for res in results]
 
+    tool_request_types = cast(
+        "Iterable[type[ExportToolRequest]]", union_membership.get(ExportToolRequest)
+    )
+    tool_requests = tuple(request_type() for request_type in tool_request_types)
+    tool_results = await MultiGet(
+        Get(ExportResult, ExportToolRequest, request) for request in tool_requests
+    )
+    flattened_results.extend(tool_results)
```

Comment:
Ditto that this should live in the Python file. 

We will definitely end up doing something with union membership - I am hoping that we do not need a new union base like this though, as it would imply that every Python tool must implement a rule to fulfill this plug-in hook. Instead, I am hoping that we can leverage the plug-in hook we already have for generating lock files, although I am not 100% certain yet.